From 8459fb8f8941618b3e8ac1015ebce9aa4ed878c6 Mon Sep 17 00:00:00 2001
From: Alain Volmat <alain.volmat@foss.st.com>
Date: Thu, 26 Sep 2024 15:01:32 +0200
Subject: [PATCH] Revert "media: ov5640: Set JPEG output timings when
 outputting JPEG data"

This reverts commit 7cb013b1049a2a599c1fc344644528eb8bf985b8.

Revert JPEG related tuning of the sensor since this leads to
regressions such as framerate drop in high resolutions (> 720p)
and corrupted data.

Change-Id: I8aff31f0c7ef874403d499180788580cb11b4397
Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/407143
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Hugues FRUCHET <hugues.fruchet@st.com>
Reviewed-by: Alain VOLMAT <alain.volmat@st.com>
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
Tested-by: Alain VOLMAT <alain.volmat@st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
---
 drivers/media/i2c/ov5640.c | 21 ---------------------
 1 file changed, 21 deletions(-)

diff --git a/drivers/media/i2c/ov5640.c b/drivers/media/i2c/ov5640.c
index 191351506579..af3cc7e61da9 100644
--- a/drivers/media/i2c/ov5640.c
+++ b/drivers/media/i2c/ov5640.c
@@ -103,8 +103,6 @@
 #define OV5640_REG_SIGMADELTA_CTRL0C	0x3c0c
 #define OV5640_REG_FRAME_CTRL01		0x4202
 #define OV5640_REG_FORMAT_CONTROL00	0x4300
-#define OV5640_REG_VFIFO_HSIZE		0x4602
-#define OV5640_REG_VFIFO_VSIZE		0x4604
 #define OV5640_REG_CCIR656_CTRL00	0x4730
 #define OV5640_REG_POLARITY_CTRL00	0x4740
 #define OV5640_REG_MIPI_CTRL00		0x4800
@@ -1652,19 +1650,6 @@ static int ov5640_set_dvp_pclk(struct ov5640_dev *sensor)
 			      (ilog2(pclk_div) << 4));
 }
 
-/* set JPEG framing sizes */
-static int ov5640_set_jpeg_timings(struct ov5640_dev *sensor,
-				   const struct ov5640_mode_info *mode)
-{
-	int ret;
-
-	ret = ov5640_write_reg16(sensor, OV5640_REG_VFIFO_HSIZE, mode->width);
-	if (ret < 0)
-		return ret;
-
-	return ov5640_write_reg16(sensor, OV5640_REG_VFIFO_VSIZE, mode->height);
-}
-
 /* download ov5640 settings to sensor through i2c */
 static int ov5640_set_timings(struct ov5640_dev *sensor,
 			      const struct ov5640_mode_info *mode)
@@ -1674,12 +1659,6 @@ static int ov5640_set_timings(struct ov5640_dev *sensor,
 	const struct v4l2_rect *crop;
 	int ret;
 
-	if (sensor->fmt.code == MEDIA_BUS_FMT_JPEG_1X8) {
-		ret = ov5640_set_jpeg_timings(sensor, mode);
-		if (ret < 0)
-			return ret;
-	}
-
 	timings = ov5640_timings(sensor, mode);
 	analog_crop = &timings->analog_crop;
 	crop = &timings->crop;
-- 
2.39.5

